### DISCUSSION TITLE
Is it possible to use `where` with the Content Query API (using Endpoints)?

### DISCUSSION THREAD

**Discussion by nobkd** [nuxt/content#1420] [Q&A] [answered]
I wanted to use the [API routes](https://content.nuxtjs.org/guide/displaying/querying/#api-routes) with a `where` query like using `queryContent()`. Is that possible?

If it's possible, I'd like a simple example like:
`queryContent().where({ _path: {$contains: 'a'}}).find()` translated to
`$fetch('/api/_content/query?where=___')` or how it works.

**Comment by yungParrot** [comment, marked as answer]
The docs don't specify how it should be done... After looking at the source code of [the query module](https://github.com/nuxt/content/blob/main/src/runtime/utils/query.ts#L28), I've figured out that you could send JSON parameters to the API 👇 

```typescript
const query = {
  where: { _path: { $contains: "a" } }
};

const response = axios.get(`/api/_content/query?_params=${JSON.stringify(query)}`);
```

This should return all articles containing `"a"` in their `_path`.

👉 https://content.nuxtjs.org/api/composables/query-content

---

🤠

**Reply by nobkd** [reply]
Thank you 🙏 

**EDIT:** If someone wants to write the query without using the `JSON.stringify(query)` method, the route for the request looks like this:
```
/api/_content/query?_params={"where":{"_path":{"$contains":"a"}}}
```
=> The keys and values are wrapped in quotes
- Regex querys(+flags) are also wrapped in quotes: `{"where":{"_path":{"$regex":"/a$/i"}}}` => returns pages whose paths end on `a` or `A`
- regex(+flags) have to be wrapped in quotes in the **not stringified object notation** too
- It's mandatory to use double quotes `"`, single quotes `'` **won't work**
